cmake/checks/check_01_cxx_features.cmake

```diff
@@ -753,6 +754,19 @@ CHECK_CXX_SOURCE_COMPILES(
   "
   DEAL_II_HAVE_CXX14_CONSTEXPR_CAN_CALL_NONCONSTEXPR)
 
+#
+# The macro DEAL_II_CONSTEXPR allows using c++ constexpr features in a portable way.
+# Here we enable it only when a constexpr function can call simple non-constexpr
+# functions. This requirement is probabely very conservative in most cases, but
+# it will prevent breaking builds with certain compilers.
+#
+IF (DEAL_II_HAVE_CXX14_CONSTEXPR_CAN_CALL_NONCONSTEXPR)
+  MESSAGE(STATUS "Enabling CXX14 constexpr features")
```

Comment:
Let's remove the status message here. It is a minor detail and will be present in the `config.h` file in case someone wants to look it up.